transactions/transactions_service.go

```diff
@@ -730,6 +786,68 @@ func (svc *transactionsService) ConsumeEvent(ctx context.Context, event *events.
 			}).WithError(err).Error("Failed to execute DB transaction")
 			return
 		}
+
+	case "nwc_lnclient_hold_invoice_accepted":
+		lnClientTransaction, ok := event.Properties.(*lnclient.Transaction)
+		if !ok {
+			logger.Logger.WithField("event", event).Error("Failed to cast event properties for hold invoice accepted")
+			return
+		}
+		logger.Logger.WithFields(logrus.Fields{
+			"paymentHash": lnClientTransaction.PaymentHash,
+		}).Info("Processing hold invoice accepted event")
+
+		var dbTransaction db.Transaction
+		err := svc.db.Transaction(func(tx *gorm.DB) error {
+			result := tx.Where("payment_hash = ? AND type = ? AND state = ?", lnClientTransaction.PaymentHash, constants.TRANSACTION_TYPE_INCOMING, constants.TRANSACTION_STATE_PENDING).First(&dbTransaction)
+			if result.Error != nil {
+				if errors.Is(result.Error, gorm.ErrRecordNotFound) {
+					logger.Logger.WithFields(logrus.Fields{
+						"paymentHash": lnClientTransaction.PaymentHash,
+					}).Warn("No corresponding pending incoming transaction found in DB for accepted hold invoice")
+					var existingTx db.Transaction
+					if tx.Where("payment_hash = ? AND type = ? AND (state = ? OR state = ?)", lnClientTransaction.PaymentHash, constants.TRANSACTION_TYPE_INCOMING, constants.TRANSACTION_STATE_ACCEPTED, constants.TRANSACTION_STATE_SETTLED).First(&existingTx).Error == nil {
+						logger.Logger.WithFields(logrus.Fields{
+							"paymentHash":  lnClientTransaction.PaymentHash,
+							"currentState": existingTx.State,
+						}).Info("Transaction already in accepted/settled state")
+						return nil // Not an error, already handled
+					}
+				}
+				logger.Logger.WithFields(logrus.Fields{
+					"paymentHash": lnClientTransaction.PaymentHash,
+				}).WithError(result.Error).Error("Failed to query DB for accepted hold invoice")
+				return result.Error
+			}
+
+			err := tx.Model(&dbTransaction).Update("state", constants.TRANSACTION_STATE_ACCEPTED).Error
+			if err != nil {
+				logger.Logger.WithFields(logrus.Fields{
+					"paymentHash": lnClientTransaction.PaymentHash,
+					"dbTxID":      dbTransaction.ID,
+				}).WithError(err).Error("Failed to update hold invoice state to accepted in DB")
+				return err
+			}
+
+			logger.Logger.WithFields(logrus.Fields{
+				"paymentHash": lnClientTransaction.PaymentHash,
+				"dbTxID":      dbTransaction.ID,
+			}).Info("Updated hold invoice state to accepted in DB")
+
+			var updatedDbTransaction db.Transaction
+			tx.First(&updatedDbTransaction, dbTransaction.ID)
+			svc.eventPublisher.Publish(&events.Event{
```

Comment:
can publishing be moved out of the DB transaction, otherwise this will lock the entire DB